db/upgradelib.php

```diff
@@ -69,25 +69,32 @@ function surveypro_delete_supposed_blank_answers() {
     // Parent items without a saved answer must be considered as parent NOT allowing children.
     // Delete all the answers to children items that have parent without saved answers.
     // Alias: if the parent was not allowed in the userform, its children will not be allowed even more.
-    $sql = 'SELECT id as childid, parentid, surveyproid
+    $sql = 'SELECT parentid, MAX(surveyproid) as surveyproid
             FROM {surveypro_item}
             WHERE type = :type
                 AND parentid <> :parentid
             GROUP BY parentid
-            ORDER BY surveyproid, sortindex';
+            ORDER BY 2';
     $whereparams = array('type' => SURVEYPRO_TYPEFIELD, 'parentid' => 0);
     $parentitems = $DB->get_records_sql($sql, $whereparams);
 
+    $oldsurveyproid = 0;
     foreach ($parentitems as $parentitem) {
         // Get all submissions for this surveyproid.
-        $whereparams = array();
-        $whereparams['surveyproid'] = $parentitem->surveyproid;
-        $submissions = $DB->get_recordset('surveypro_submission', $whereparams, 'id', 'id');
+        if ($parentitem->surveyproid != $oldsurveyproid) {
+            $whereparams = array();
+            $whereparams['surveyproid'] = $parentitem->surveyproid;
+            $submissions = $DB->get_recordset('surveypro_submission', $whereparams, 'id', 'id');
+
+            $oldsurveyproid = $parentitem->surveyproid;
+        }
         foreach ($submissions as $submission) {
```

Comment:
mmmhhhh, why do you say this?

At first round I am sure I get in:
if ($parentitem->surveyproid != $oldsurveyproid) {
(because surveyprois can't be == 0) and I get:
$submissions = $DB->...

Next rounds, if I get in the if {I change the set} else {the old set of submissions is still correct}